### PR TITLE
Unit tests for shuffler

### PR DESCRIPTION
**Pull request by krfricke**

### CODE REVIEW

File: ray_shuffling_data_loader/tests/test_shuffle.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        parts_per_shuffler = num_reducers // num_shufflers
+        for i in range(num_shufflers):
+            unshuffled_refs = fetched_refs[(i * parts_per_shuffler):(
+                i + 1 * parts_per_shuffler)]
+            unshuffled_parts = fetched_parts[(i * parts_per_shuffler):(
+                i + 1 * parts_per_shuffler)]
+
+            shuffled = ray.get(
+                shuffle_reduce.remote(
+                    0,
```
Comment (clarkzinzow) [inline ray_shuffling_data_loader/tests/test_shuffle.py:98]: It shouldn't matter much (I actually don't think it's even used in the reducer anymore), but maybe we should set the `reducer_index` here.
```suggestion
                    i,
```

File: ray_shuffling_data_loader/tests/test_shuffle.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+            "Trainer count mismatch"
+
+        assert all(len(consumer.rank_epoch_batches[t]) == num_epochs
+                   for t in consumer.rank_epoch_batches), \
+            "Epoch count mismatch"
+
+        for tid, epoch_batches in consumer.rank_epoch_batches.items():
+            for i in range(len(epoch_batches) - 1):
+                assert len(epoch_batches[i]) == len(
+                    epoch_batches[+1]) == num_epochs, \
```
Comment (clarkzinzow) [inline ray_shuffling_data_loader/tests/test_shuffle.py:151]:
```suggestion
                    epoch_batches[1]) == num_epochs, \
```

File: ray_shuffling_data_loader/tests/test_shuffle.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+                set2 = set(keys2)
+
+                assert len(set1) == len(keys1), \
+                    "Keys in dataset are not distinct."
+
+                assert len(set2) == len(keys2), \
+                    "Keys in dataset are not distinct."
+
+                assert set1 == set2, \
+                    "Shuffled key sets are not equal."
```
Comment (clarkzinzow) [inline ray_shuffling_data_loader/tests/test_shuffle.py:170]: Great e2e test!

**Review by clarkzinzow — APPROVED** [review]
LGTM overall, just a few nits and question.
